src/Parser/DestructionWarlock/Modules/Features/AlwaysBeCasting.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import CoreAlwaysBeCasting from 'Parser/Core/Modules/AlwaysBeCasting';
+
+import SPELLS from 'common/SPELLS';
+import Icon from 'common/Icon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import SpellLink from 'common/SpellLink';
+
+class AlwaysBeCasting extends CoreAlwaysBeCasting {
+  static ABILITIES_ON_GCD = [
+    SPELLS.CHAOS_BOLT.id,
+    SPELLS.CONFLAGRATE.id,
+    SPELLS.DEMONIC_GATEWAY_CAST.id,
+    SPELLS.DRAIN_LIFE.id,
+    SPELLS.HAVOC.id,
+    SPELLS.HEALTH_FUNNEL_CAST.id,
+    SPELLS.IMMOLATE_CAST.id,
+    SPELLS.INCINERATE.id,
+    SPELLS.LIFE_TAP.id,
+    SPELLS.SOULSTONE.id,
+    SPELLS.SUMMON_DOOMGUARD_UNTALENTED.id,
+    SPELLS.SUMMON_INFERNAL_UNTALENTED.id,
+    SPELLS.GRIMOIRE_IMP.id,
+    SPELLS.UNENDING_RESOLVE.id,
+    //talents
+    SPELLS.BURNING_RUSH.id,
+    SPELLS.CATACLYSM.id,
+    SPELLS.CHANNEL_DEMONFIRE_CAST.id,
+    SPELLS.DEMONIC_CIRCLE_SUMMON.id,
+    SPELLS.DEMONIC_CIRCLE_TELEPORT.id,
+    SPELLS.MORTAL_COIL.id,
+    SPELLS.SOUL_HARVEST.id,
+    SPELLS.SHADOWBURN.id,
+    //practically unused, for the sake of completeness
+    SPELLS.SUMMON_DOOMGUARD_TALENTED.id,
+    SPELLS.SUMMON_INFERNAL_TALENTED.id,
+    SPELLS.DARK_PACT.id,
+    SPELLS.GRIMOIRE_OF_SACRIFICE_BUFF.id,
+    SPELLS.SHADOWFURY_TALENT.id,
+    SPELLS.GRIMOIRE_VOIDWALKER.id,
+    SPELLS.GRIMOIRE_FELHUNTER.id,
+    SPELLS.GRIMOIRE_SUCCUBUS.id,
+  ];
+
+  on_initialized() {
```

Comment:
This does nothing and could be omitted.

Reply:
Didn't see it in the first place, same problem will probably be in Affliction ABC too (I copied), will delete.